src/pages/api/donation-settings/update/[id].ts

```diff
@@ -0,0 +1,50 @@
+import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { updateDonationSettings } from "~/lib/db/donationSettings";
+import { withSessionRoute } from "~/lib/withSession";
+
+const handler: NextApiHandler = async (req, res) => {
+  switch (req.method) {
+    case "POST":
+      updateStreamerSettings(req, res);
+      break;
+    default:
+      res.setHeader("Allow", ["POST"]);
+      res.status(405).end(`Method ${req.method} Not Allowed`);
+  }
+};
+
+async function updateStreamerSettings(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  const user = request.session.user;
+
+  if (!user || user.isLoggedIn === false) {
```

Comment:
We need some kind of reusable authentication middleware to protect private endpoints e.g. isLoggedIn